sdk/identity/identity/src/msal/errors.ts

```diff
@@ -0,0 +1,9 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { CredentialUnavailable } from "../client/errors";
+
+/**
+ * Error used to enforce authentication after trying to retrieve a token silently.
+ */
+export class AuthenticationRequired extends CredentialUnavailable {}
```

Comment:
should set the error name field, since I don't think this happens automatically. e.g. https://github.com/Azure/azure-sdk-for-js/blob/6689ce97063c5af82f83e19ea7b107d1df96ceda/sdk/core/core-rest-pipeline/src/restError.ts#L71

Reply:
`AuthenticationRequiredException` needs to have the details of the authentication request. In .NET this is a `TokenRequestContext` instance, for JS the  analog would be requested `scopes` and `GetTokenOptions` I believe.